### PR TITLE
Delete rcl_impl_getenv, replaced by rcutils_get_env

### PR DESCRIPTION
**Pull request by ivanpauno**
Just what the title says.

### CODE REVIEW

**Comment by clalancette** [comment]
FYI: this fixes #119

**Comment by ivanpauno** [comment]
mmm, so this wasn't moved before because `rcutils_get_env` isn't thread safe, right?
We're currently using `rcutils_get_env` in other places like the different rmw implementations, without caring about thread safety. We should also solve that problem in a different PR (maybe, before merging this one).

**Comment by ivanpauno** [comment]
> maybe, before merging this one

Well, `getenv` itself is not thread safe, so this change could be merged independently of https://github.com/ros2/rcutils/pull/182.

File: rcl/src/rcl/rmw_implementation_identifier_check.c [hunk 1]
```diff
@@ -59,14 +62,14 @@ INITIALIZER(initialize) {
   rcl_allocator_t allocator = rcl_get_default_allocator();
   char * expected_rmw_impl = NULL;
   const char * expected_rmw_impl_env = NULL;
-  rcl_ret_t ret = rcl_impl_getenv("RMW_IMPLEMENTATION", &expected_rmw_impl_env);
-  if (ret != RCL_RET_OK) {
+  const char * get_env_error_str = NULL;
+  get_env_error_str = rcutils_get_env(RMW_IMPLEMENTATION_ENV_VAR_NAME, &expected_rmw_impl_env);
```
Comment (hidmic) [inline rcl/src/rcl/rmw_implementation_identifier_check.c]: @ivanpauno nit: collapse the last two lines into one.

**Comment by ivanpauno** [comment]
@hidmic Addressed your only comment, and also rebased with master.

**Comment by ivanpauno** [comment]
CI:
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/8200/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/4175/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/6678/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/8074/)

**Pull request merged**